Convert life360 to aiohttp

pnbruckner opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Life360 (the pypi package, and hence, the HA integration as well) has always used the
`requests` package and, so, has had to be run in a separate executor thread. This change
updates to v5.0.0 of the life360 package which now uses `aiohttp`, and modifies the
HA integration accordingly.

This does not really improve the speed at which data can be retrieved from the server,
but it should reduce the overhead associated with this network traffic.

Changes to `life360` package:
- https://github.com/pnbruckner/life360/compare/v4.1.1...v5.1.0

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

pnbruckner committed:
Convert life360 from requests to aiohttp

Bump life360 package to 5.0.0.

IceBotYT commented on homeassistant/components/life360/config_flow.py:
```diff
@@ -69,10 +67,16 @@ def async_get_options_flow(config_entry: ConfigEntry) -> Life360OptionsFlow:
         """Attempt to authorize the provided credentials."""
+        if not self._api:
+            self._api = Life360(
+                session=async_create_clientsession(
```
Shouldn't `async_get_clientsession` be used instead?

pnbruckner replied:
Ah, yes, I think you're right. I used `async_create_clientsession` in coordinator.py because a `ConfigEntry` is associated with it, but here in config_flow.py that's not the case. I guess I'll need to rev the `life360` package and make it specify the timeout in the request as opposed to having it set as a default in the session.

pnbruckner replied:
Actually, I probably should change the call in coordinator.py, too. It doesn't require a unique session (e.g., it doesn't need its own separate set of cookies), so `async_get_clientsession` should work just fine there as well. I'll add that before pushing a new commit.

pnbruckner committed:
Update per review

Use async_get_clientsession instead of async_create_clientsession.

Bump life360 package to 5.1.0.

Pass timeout to Life360 init instead of adding to session.

pnbruckner commented:
Not sure why there are test failures all of a sudden, seemingly unrelated. I'm rebasing and I'll force push an update once local tests pass.

IceBotYT reviewed (approved):
LGTM! 👍
